### PR TITLE
Add api-diff agent skill

### PR DESCRIPTION
**Pull request by aruniverse**
This pull request introduces a new `api-diff` skill for comparing public API changes between two branches or version tags in the iTwin.js monorepo. It provides a Bash script for diffing API files, a comprehensive skill documentation, a reference on API support and breaking change policies, and a test script to ensure correct behavior. The main improvements are structured API diffing, changelog coverage checking, and robust testability.

The most important changes are:

**Skill and Policy Documentation**
- Added `.agents/skills/api-diff/SKILL.md` with detailed instructions, usage scenarios, categorization of API changes, and output formatting for the new API diff skill.
- Added `.agents/skills/api-diff/REFERENCE.md` summarizing the iTwin.js API support policy, including tag meanings, promotion paths, deprecation rules, and what constitutes a breaking change.

**API Diff Script**
- Added `.agents/skills/api-diff/scripts/api-diff.sh`, a Bash script that compares `common/api/*.api.md` files between two git refs, summarizes API changes per package, and shows relevant changelog headings. Handles branch/tag naming conventions, large diffs, and changelog range computation.

**Testing**
- Added `.agents/skills/api-diff/scripts/test-api-diff.sh` to provide unit and integration tests for the diff script, including version parsing, changelog filtering, and a smoke test. Ensures reliability and correctness of the diff logic.

### CODE REVIEW

File: .agents/skills/api-diff/SKILL.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+Tests cover `ref_to_minor_ver` parsing, Python changelog version filtering (boundary conditions, numeric sorting), and a smoke test of the full script. Run after any changes to the script.
+
+## Tips
+
+- Ignore `common/api/summary/*.exports.csv` — generated metadata only.
+- All packages use **lockstep versioning** — one version number for all packages.
+- `@public @preview` ≠ `@beta`. Preview APIs are production-ready but may change in the next major version (3-month removal grace period vs 1-year for plain `@public`).
+- Missing release tags (`// (undocumented)`) are treated as `@internal`.
+- `git diff release/5.9.0 release/5.10.0 -- common/api/core-backend.api.md | grep "^[+-]" | grep -v "^[+-][+-]"` for a quick per-file scan.
```
Comment (ben-polinsky) [inline .agents/skills/api-diff/SKILL.md:93]: One practical issue here is that the docs point people at release branch names like `release/5.10.0`, but this repo's release branches are `release/X.Y.x`.

That makes the quick-start examples and the workflow table fail for the common release-branch case, even though `release/5.10.x` exists. I think it would help to update the examples and branch-naming text to use `release/5.10.x`, and mention tags separately if tag-based comparisons are also meant to work.

🐻🤖 gentlebenbot (powered by GPT-5.4)

File: .agents/skills/api-diff/scripts/api-diff.sh [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+    echo "========================================="
+    echo "Package: $PKG"
+    echo "========================================="
+
+    # Run git diff once per file; reuse the cached output for counts and display.
+    git --no-pager diff "$BASE" "$HEAD" -- "$f" > "$DIFF_TMP"
+
+    local RAW_ADDED RAW_REMOVED
+    RAW_ADDED=$(safe_grep "^+" "$DIFF_TMP" | safe_grep -v "^+++" \
+      | safe_grep -v "^+import " | safe_grep -v "^+[[:space:]]*//" | wc -l | tr -d ' ')
```
Comment (ben-polinsky) [inline .agents/skills/api-diff/scripts/api-diff.sh:175]: The main thing I wanted to flag is that this filter drops the metadata lines that actually carry release tags and deprecation markers in `common/api/*.api.md` files.

Right now both the count and display paths exclude added and removed `// ...` lines, which means the script cannot surface changes like `@beta -> @public`, `@public @preview -> @public`, or newly added `@deprecated` markers even though the skill documentation says it does. In this repo, those lines are the compatibility signal.

I think the safest fix here is to keep those metadata lines in the diff output, or parse them separately before filtering, so tag promotions and deprecations do not disappear from the report.

🐻🤖 gentlebenbot (powered by GPT-5.4)
